api/api-member/src/main/java/com/pgms/apimember/service/MemberService.java

```diff
@@ -0,0 +1,100 @@
+package com.pgms.apimember.service;
+
+import static com.pgms.apimember.exception.CustomErrorCode.*;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.pgms.apimember.dto.request.MemberInfoUpdateRequest;
+import com.pgms.apimember.dto.request.MemberPasswordUpdateRequest;
+import com.pgms.apimember.dto.response.MemberDetailGetResponse;
+import com.pgms.apimember.exception.CustomErrorCode;
+import com.pgms.apimember.exception.MemberException;
+import com.pgms.coredomain.domain.member.Member;
+import com.pgms.coredomain.domain.member.repository.MemberRepository;
+
+import lombok.RequiredArgsConstructor;
+
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class MemberService {
+
+	private final MemberRepository memberRepository;
+
+	@Transactional(readOnly = true)
+	public MemberDetailGetResponse getMemberDetail(Long memberId) {
+		return MemberDetailGetResponse.from(
+			memberRepository.findByIdWithRole(memberId).
+				orElseThrow(() -> new MemberException(CustomErrorCode.MEMBER_NOT_FOUND)));
+	}
+
+	@Transactional(readOnly = true)
+	public void verifyPassword(Long memberId, String password) {
+		final Member member = getAvailableMember(memberId);
+		validatePlainPassword(password, member.getPassword());
+	}
+
+	public void updateMember(Long memberId, MemberInfoUpdateRequest requestDto) {
+		final Member member = getAvailableMember(memberId);
+		member.updateMemberInfo(
+			requestDto.name(),
+			requestDto.phoneNumber(),
+			requestDto.birthDate(),
+			requestDto.gender(),
+			requestDto.streetAddress(),
+			requestDto.detailAddress(),
+			requestDto.zipCode()
+		);
+	}
+
+	public void updatePassword(Long memberId, MemberPasswordUpdateRequest requestDto) {
+		final Member member = getAvailableMember(memberId);
+		validatePlainPassword(requestDto.originPassword(), member.getPassword());
+		validatePasswordAndConfirm(requestDto.newPassword(), requestDto.newPasswordConfirm());
+		member.updatePassword(requestDto.newPassword()); // TODO: 비밀번호 암호화 추가 필요
+
+	}
+
+	public void deleteMember(Long memberId) {
+		final Member member = getAvailableMember(memberId);
+		member.updateToDeleted();
+	}
+
+	public Long restoreMember(Long memberId) {
+		final Member member = memberRepository.findById(memberId)
+			.orElseThrow(() -> new MemberException(MEMBER_NOT_FOUND));
+
+		if (member.isDeleted()) {
+			member.updateToActive();
+		}
```

Comment:
```suggestion
                Member member = memberRepository.findByIdAndIsDeletedTrue(memberId)
                    .orElseThrow(() -> new MemberException(MEMBER_NOT_FOUND));
            
                member.updateToActive();
```

여기도 개인적인 코드 스타일 의견 제안 드립니당 ㅎㅎ

Reply:
감사합니다!! 이런식으로 하는게 더 깔끔할 것 같네요!
리뷰 내용 적용 후 머지하겠습니다!